### PR TITLE
improved code coverage of Donate.tsx

### LINKED ISSUE #3038: Improve Code Coverage in `src/screens/UserPortal/Donate/Donate.tsx`
# Goal  
The goal of this issue is to improve the code coverage for the screen: `src/screens/UserPortal/Donate/Donate.tsx`

# Tasks  
1. Review the file to identify sections of code that are being ignored by Codecov or are not covered by tests.  
2. Create or update test cases to ensure 100% coverage for the file.  
3. Remove any `/* istanbul ignore */` or equivalent statements that bypass code coverage reporting, unless absolutely necessary.  

# Resources  
- Refer to the foundational documentation on writing test cases in the repository.  
- Check the [Codecov report](<link-to-codecov-report>) for details on the uncovered lines.  

# Acceptance Criteria  
- All sections of the file are covered by tests.  
- Code coverage for the file reaches 100%.  
- PR created with necessary updates, passing all checks and reviews.

### PR DESCRIPTION
**Pull request by Dhiren-Mhatre**
**What kind of change does this PR introduce?**

Improved Code Coverage in src/screens/UserPortal/Donate/Donate.tsx

**Issue Number:**

Fixes #3038

**Did you add tests for your changes?**

Yes. Added comprehensive test coverage for `src/screens/UserPortal/Donate/Donate.tsx`

**Snapshots/Videos:**

https://github.com/user-attachments/assets/ba67d35f-b3e5-4691-9b5d-ac4cac7e2ad1

**If relevant, did you update the documentation?**

N/A

**Summary**

This PR improves code coverage for the Donate component by:
- Adding test cases for previously uncovered code paths
- Removing unnecessary istanbul ignore statements
- Ensuring 100% test coverage of the component

**Does this PR introduce a breaking change?**

No

**Other information**

Code coverage increased from previous coverage to 100% for the specified component.

**Have you read the contributing guide?**

Yes
 

## Summary by CodeRabbit

- **New Features**
  - Enhanced donation process with improved asynchronous handling
  - Added test coverage for donation interactions

- **Bug Fixes**
  - Improved error handling for donation mutations
  - Refined pagination and loading state management

- **Tests**
  - Expanded test suite for donation component
  - Added comprehensive test cases for user interactions and error scenarios

### CODE REVIEW

File: src/screens/UserPortal/Donate/Donate.spec.tsx [hunk 1]
```diff
@@ -188,6 +194,237 @@ describe('Testing Donate Screen [User Portal]', () => {
...
+              <Donate />
+            </I18nextProvider>
+          </Provider>
+        </BrowserRouter>
+      </MockedProvider>,
+    );
+
+    const amountInput = screen.getByPlaceholderText('Amount');
+    fireEvent.change(amountInput, { target: { value: '100' } });
+    expect(amountInput).toHaveValue('100');
```
Comment (beingnoble03) [inline src/screens/UserPortal/Donate/Donate.spec.tsx:212]: we are not checking the state here (as the test description says)

Comment (beingnoble03) [reply src/screens/UserPortal/Donate/Donate.spec.tsx:212]: Sorry, I see why it's tested that way!

**Review by beingnoble03 — APPROVED** [review]
LGTM!

File: src/screens/UserPortal/Donate/Donate.spec.tsx (hunk 1 above)
Comment (Dhiren-Mhatre) [reply src/screens/UserPortal/Donate/Donate.spec.tsx:212]: Thanks for your feedback!

**Pull request merged**